File: python/2nd_week/Praktikum2/latihanpraktikum2_python/Memory.py
from Hardware import Hardware

class Memory(Hardware):

    def __init__(self):
        self.frequency = ""
        self.memorySize = ""
        self.supportsCuda = False

    def setFreqeuncy(self, frequency):
        self.frequency = frequency

    def setMemorySize(self, memorySize):
        self.memorySize = memorySize

    def setSupportsCuda(self, supportsCuda):
        self.supportsCuda = supportsCuda

    def getFrequency(self):
        return self.frequency
    
    def getMemorySize(self):
        return self.memorySize
    
    def getSupportsCuda(self):
        return self.supportsCuda
    

File: php/2nd_week/Praktikum2/latihanpraktikum2_php/Memory.php
<?php

class Memory extends Hardware{
    private $frequency = "";
    private $memorySize = "";
    private $supportsCuda = "";

    function __construct()
    {
    }

    function setFrequency($frequency){
        $this->frequency = $frequency;
    }

    function setMemorySize($memorySize){
        $this->memorySize = $memorySize;
    }

    function setSupportsCuda($supportsCuda){
        $this->supportsCuda = $supportsCuda;
    }

    function getFrequency(){
        return $this->frequency;
    }

    function getMemorySize(){
        return $this->memorySize;
    }

    function getSupportsCuda(){
        return $this->supportsCuda;
    }

    function __destruct()
    {
    }
}
?>